go/common/identity/identity.go

```diff
@@ -366,21 +173,12 @@ func doLoadOrGenerate(dataDir string, signerFactory signature.SignerFactory, sho
 		P2PSigner:                  signers[1],
 		ConsensusSigner:            signers[2],
 		VRFSigner:                  signers[3],
-		tlsSigner:                  memory.NewFromRuntime(cert.PrivateKey.(ed25519.PrivateKey)),
-		tlsCertificate:             cert,
-		nextTLSSigner:              nextSigner,
-		nextTLSCertificate:         nextCert,
-		DoNotRotateTLS:             dontRotate,
+		TLSSigner:                  memory.NewFromRuntime(cert.PrivateKey.(ed25519.PrivateKey)),
+		TLSCertificate:             cert,
 		TLSSentryClientCertificate: sentryClientCert,
-		tlsRotationNotifier:        pubsub.NewBroker(false),
-		dataDir:                    dataDir,
 	}, nil
 }
 
-func ephemeralKeyPath(dataDir, generation string) string {
-	return filepath.Join(dataDir, fmt.Sprintf("%s%s.pem", tlsEphemeralKeyBaseFilename, generation))
```

Comment:
Do we need to remove these files, to not pollute the datadir?

Reply:
I would suggest removing them as part of `doLoadOrGenerate` in case they exist.

Reply:
Agree. I would also add a TODO comment and prepare a draft pull request to remove this cleanup code immediately after the next release.